recipes/libcurand/meta.yaml

```
@@ -0,0 +1,102 @@
+{% set name = "libcurand" %}
+{% set version = "10.3.1.50" %}
+{% set cuda_version = "12.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 7c25712016fdde96011c3370673d9007eddd7231261a9b47f36b213ceb964191  # [linux64]
+  sha256: ac020f6e88105927f96f58ca331d3cddba35b37376f4716e866155c4f56681dd  # [ppc64le]
+  sha256: c7ae331dd6c9936a4e9746abdc222514bbda47a1e495e396b8244b918de377e2  # [aarch64]
+  sha256: 5fb95fbbd1c652e1e2a8c12c41d225117cab3f718c739f0d49d580c11b4a44f1  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+
+outputs:
+  - name: libcurand
+    files:
+      - lib/libcurand.so.*            # [linux]
+      - targets/*/lib/libcurand.so.*  # [linux]
+      - Library\bin\curand*.dll       # [win]
+    requirements:
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcurand.so.{{ version }}                              # [linux]
+        - if not exist %LIBRARY_BIN%\curand64_{{ version.split(".")[0] }}.dll exit 1  # [win]
+
+  - name: libcurand-dev
```

Suggesting so that `libcurand` is added as a dependency automatically in `run` when using this package in `host`. This [aligns with `libcublas-dev`]( https://github.com/conda-forge/staged-recipes/pull/21901/files#diff-dfb8a16c4242dd60a2768d93ec54e02b92fe4b0cd252f2cfcbecc8b0d79e2c91R50-R52 )

cc @bdice

```suggestion
  - name: libcurand-dev
    build:
      run_exports:
        - {{ pin_subpackage("libcurand", max_pin="x") }}
```